packages/flutter_tools/lib/src/cache.dart

```diff
@@ -434,7 +446,9 @@ class Cache {
     );
   }
 
-  bool isUpToDate() => _artifacts.every((ArtifactSet artifact) => artifact.isUpToDate());
+  FutureOr<bool> isUpToDate() async => (await Future.wait<bool>(<Future<bool>>[
+    ..._artifacts.map((ArtifactSet artifact) async => await artifact.isUpToDate()),
```

Comment:
This new version is pretty dense. Why do you need to destructure `...` the output of `.map()` here, I would assume that the `.map()` is returning a `List<Future<bool>>` already?

Reply:
I had to do this because `FutureOr<bool>` is not a `Future<bool>`, so I can't pass it to Future.wait. I should just change to type signature to skip the FutureOr, then I can simplify this

Reply:
Ahh, that makes sense. Is `Future<bool>` a sub-type of `FutureOr<bool>`?